### PR TITLE
Use pipeline artifacts in CI

### LINKED ISSUE #101: Add windows agent to our build and run test-cli

### PR DESCRIPTION
**Pull request by carolynvs**
# What does this change
* save the bin directory after cross compiling
* split the build into stages: doc check, build, test, smoke test
* pass the bin directory to latest stages instead of rebuilding
* run the smoke tests against windows, linux and macos

# What issue does it fix
Cross OS testing on pull requests
Closes #101 

# Notes for the reviewer
N/A

# Checklist
- [ ] Unit Tests
- [ ] Documentation
- [ ] Schema (porter.yaml)

### CODE REVIEW

**Comment by carolynvs** [comment]
@vdice  If it's okay I will do a follow-up PR to the contributing guide. I'd love to get this in, stop rebasing and then iterate on the docs which will skip the build and go much faster.

**Review by vdice — APPROVED** [review]
LGTM!  Very excited for mage 🧙‍♀️ ✨ !

File: build/azure-pipelines.release.yml [hunk 1]
```diff
@@ -59,7 +59,7 @@ stages:
       displayName: 'Build'
 
     - script: |
-        make test-cli
+        make teste2e
```
[the remark this answers is not in the document]
Comment (carolynvs) [reply build/azure-pipelines.release.yml:62]: oh goodness, I was so focused on the one build I forgot to update the other. Lemme fix.

**Comment by carolynvs** [comment]
FYI the test failure is due to 503 errors from our CDN again. I'm going to try switching to cdn2 tonight.

**Comment by carolynvs** [comment]
bitnami just released a new version of the wordpress chart that caused the build to fail. I'm pinning wordress to the last chart for helm v2 to get them to pass.

**Comment by carolynvs** [comment]
@vdice OMG, don't move a muscle. I got past the build flakes, stable chart repo deprecation, and dangers foretold to make it to ~the goblin castle~ two green builds.

Anything problems that I can move to a follow-up, I will. 😅

**Comment by vdice** [comment]
@carolynvs 😯 ❗Woo hoo!! 👏 👏 👏

**Review by vdice — APPROVED** [review]

**Pull request merged**